db/db_batch.go

```diff
@@ -228,16 +238,21 @@ func (cb *cachedBatch) Clear() {
 	defer cb.lock.Unlock()
 	cb.cache = nil
 	cb.cache = make(map[hash.CacheHash][]byte)
+	cb.deleted = make(map[hash.CacheHash]bool)
 	cb.writeQueue = nil
 }
 
 // Get retrieves a record
 func (cb *cachedBatch) Get(namespace string, key []byte) ([]byte, error) {
 	cb.lock.RLock()
 	defer cb.lock.RUnlock()
-	if v, ok := cb.cache[cb.hash(namespace, key)]; ok {
+	h := cb.hash(namespace, key)
+	if v, ok := cb.cache[h]; ok {
 		return v, nil
 	}
+	if _, ok := cb.deleted[h]; ok {
+		return nil, ErrAlreadyDeleted
```

Comment:
Should we just return ErrNotExist? Make it seamless as no data pollution

Reply:
ok